src/modal/modal-container.component.ts

```diff
@@ -76,12 +77,18 @@ export class ModalContainerComponent implements OnInit, OnDestroy {
     }
   }
 
-  @HostListener('click', ['$event'])
-  onClick(event: MouseEvent): void {
+  @HostListener('mousedown', ['$event'])
+  onClickStarted(event: MouseEvent): void {
+    this.clickStartedInside = event.target !== this._element.nativeElement;
```

Comment:
Property name or condition should be changed. Actually this evals to clickStartedOutside.
Changing condition would result in
`this.clickStartedInside = event.target === this._element.nativeElement;`

Reply:
You're right, there is a misunderstanding in the naming. `this._element.nativeElement` evaluates to the backdrop, that's why if `event.target` is different, it means we didn't click on the backdrop, hence we clicked on the modal. So a more precise name would be `clickStartedInsideModalContainer` (or `clickStartedOutsideBackdrop` but it feels like a double negative)